### PR TITLE
More cooldown calls for worker processes

### PR DESCRIPTION
**Pull request by annando**
Some days ago a check was added to the worker to delay the processing when the load is too high. This check is now done at several other places. This should help to keep the system load under a given value.

### CODE REVIEW

**Comment by annando** [comment]
Hint: The license check is failing only because of the change in the daemon.php.

File: src/Core/Worker.php [hunk 1]
```diff
@@ -445,42 +445,68 @@ public static function execute(array $queue): bool
...
+		if (($load_cooldown == 0) && ($processes_cooldown == 0)) {
+			return;
 		}
 
-		$load_cooldown      = DI::config()->get('system', 'worker_load_cooldown');
-		$processes_cooldown = DI::config()->get('system', 'worker_processes_cooldown');
+		$sleeping = false;
 
-		while ((($load_cooldown > 0) || ($processes_cooldown > 0)) && ($load = System::getLoadAvg())) {
+		while ($load = System::getLoadAvg()) {
```
Comment (Quix0r) [inline src/Core/Worker.php:503]: I guess the root cause for high CPU loads are such loops:
````
<?php
while (true) {
// Do something
}
````
Now just run that little script and you will see that one CPU core is 100% busy with it. If you just add a `usleep(100);` at the end of the loop (still inside of it!) then you will see that the CPU load goes back drastically. Comment it out again and the CPU load goes up again.

**Review by Quix0r — CHANGES_REQUESTED** [review]

File: src/Core/Worker.php (hunk 1 above)
Comment (annando) [reply src/Core/Worker.php:503]: Please have a look at the loop. Later in that loop there is a `sleep(1)` - and at the end is a `break;`.

Comment (Quix0r) [reply src/Core/Worker.php:503]: Okay, I just saw the diff-view here. :-)

**Comment by MrPetovan** [comment]
PHPUnit is complaining but I cannot access `ci.friendi.ca` from my machine, the domain name isn't resolved???

**Comment by annando** [comment]
The tests are now all happy.

File: src/Core/Worker.php [hunk 2]
```diff
@@ -445,42 +445,98 @@ public static function execute(array $queue): bool
...
+	 */
+	private static function execFunction(array $queue, string $funcname, array $argv, bool $method_call)
+	{
+		$a = DI::app();
+
+		$cooldown = DI::config()->get('system', 'worker_cooldown', 0);
+		if ($cooldown > 0) {
+			Logger::notice('Pre execution cooldown.', ['cooldown' => $cooldown, 'id' => $queue['id'], 'priority' => $queue['priority'], 'command' => $queue['command']]);
+			sleep($cooldown);
+		}
```
Comment (MrPetovan) [inline src/Core/Worker.php]: Shouldn't this be included in `coolDown()`?

Comment (annando) [reply src/Core/Worker.php]: I thought about this as well. But I decided against this. Now I'm currently thinking about, if that could be an idea ...

Comment (annando) [reply src/Core/Worker.php]: Changed

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**